### PR TITLE
Fix/tao 4603 proctoring for existing deliveries

### PR DESCRIPTION
**Pull request by tikhanovichA**

### CODE REVIEW

File: model/authorization/TestTakerAuthorizationService.php [hunk 1]
```diff
@@ -87,15 +89,18 @@ public function verifyResumeAuthorization(DeliveryExecution $deliveryExecution,
     public function isProctored($deliveryId, User $user)
     {
         $propertyUri = null;
-
+        $proctoredByDefault = $this->getServiceManager()->get(ProctoredDeliveryFactoryEventsService::SERVICE_ID)->getOption('proctoredByDefault');
```
Comment (hutnikau) [inline model/authorization/TestTakerAuthorizationService.php]: Please create constant in service class for this option

File: model/authorization/TestTakerAuthorizationService.php [hunk 2]
```diff
@@ -87,15 +89,18 @@ public function verifyResumeAuthorization(DeliveryExecution $deliveryExecution,
...
         if ($deliveryId) {
             $delivery = $this->getResource($deliveryId);
             $property = $this->getProperty(ProctorService::ACCESSIBLE_PROCTOR);
             $propertyValue = $delivery->getOnePropertyValue($property);
             $propertyUri = $propertyValue ? $propertyValue->getUri() : null;
```
Comment (hutnikau) [inline model/authorization/TestTakerAuthorizationService.php]: Give this variable more descriptive name than `$property`.

File: scripts/install/OverrideDeliveryFactoryService.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ *
+ * @author Alexander Zagovorychev <zagovorichev@1pt.com>
```
Comment (hutnikau) [inline scripts/install/OverrideDeliveryFactoryService.php]: wrong author

File: config/default/DeliveryProctoring.conf.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return new oat\taoProctoring\model\execution\ProctoredDeliveryFactoryEventsService([
```
Comment (hutnikau) [inline config/default/DeliveryProctoring.conf.php]: looks like filename does not match service identifier.

File: model/execution/ProctoredDeliveryFactoryEventsService.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+namespace oat\taoProctoring\model\execution;
+
+use oat\generis\model\data\ModelManager;
+use oat\oatbox\event\Event;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\ServiceManager;
+use oat\taoProctoring\model\ProctorService;
+
+class ProctoredDeliveryFactoryEventsService extends ConfigurableService
```
Comment (hutnikau) [inline model/execution/ProctoredDeliveryFactoryEventsService.php]: In my opinion this is not EventsService. This is event handler. I believe we do not need to have this service. You could add this option and event handlers to the existing `oat\taoProctoring\model\ProctorService` service.

File: scripts/install/OverrideDeliveryFactoryService.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    public function __invoke($params)
+    {
+        $configParams['proctoredByDefault'] = true;
+        $service = new ProctoredDeliveryFactoryEventsService($configParams);
+        $service->setServiceManager($this->getServiceManager());
+        $this->getServiceManager()->register(ProctoredDeliveryFactoryEventsService::SERVICE_ID, $service);
+
+        /** @var EventManager $eventManager */
+        $eventManager = $this->getServiceManager()->get(EventManager::SERVICE_ID);
+        $eventManager->attach(DeliveryUpdatedEvent::class, [ProctoredDeliveryFactoryEventsService::class, 'update']);
```
Comment (hutnikau) [inline scripts/install/OverrideDeliveryFactoryService.php]: looks like this events will be attached only during installation of Tao. They should be attached in the updater script as well.

**Comment by tikhanovichA** [comment]
fixed

**Pull request merged**